app/scripts/services/data.js

```diff
@@ -334,27 +334,41 @@ angular.module('openshiftConsole')
     // special case, if we have an immutable item, we can return it immediately
     if (this._hasImmutable(resource, existingData, name)) {
       $timeout(function() {
+        // we can be guaranteed this wont change on us, just send what we have in existingData
         deferred.resolve(existingData.by('metadata.name')[name]);
       }, 0);
     }
     else if (!force && this._isCached(key)) {
-      var obj = existingData.by('metadata.name')[name];
-      if (obj) {
-        $timeout(function() {
-          deferred.resolve(obj);
-        }, 0);
-      }
-      else {
-        $timeout(function() {
+      $timeout(function() {
+        // re-pull the data from the cache in case it has changed on us from the watch
+        var data = this._data(key);
+        if (!data) {
+          // Extremely unlikely, would require that the watch was closed and other watches pushed
+          // this data out of the cache
```

Comment:
the alternative in this scenario is to go do the GET if we've lost the entire cached data map in the interim, but it means a significantly bigger refactor, if are going to cherry-pick this into 1.3.1 i'd feel better with a less risky refactor